tiledb/sm/array_schema/array_schema.h

```diff
@@ -297,9 +297,12 @@ class ArraySchema {
   /** Returns the number of dimensions. */
   dimension_size_type dim_num() const;
 
-  /** Dumps the array schema in ASCII format in the selected output. */
+  /** Dumps the array schema in ASCII format in the selected output file. */
   void dump(FILE* out) const;
 
+  /** Dumps the array schema in ASCII format in the selected output string. */
+  void dump(std::string* out) const;
```

Comment:
```suggestion
  std::string dump() const;
```

Same in other classes. The separate methods `dump_array_schema` and co. can be merged.

Reply:
I tried to follow the existing logic: https://github.com/TileDB-Inc/TileDB/blob/76dbda43d98bff7b71527fbffd0dfd657437b00f/tiledb/sm/stats/global_stats.cc#L69-L71

Should we also change the existing methods or not because it will cause potential backward compatibility issues for existing code?

Reply:
>the existing logic

The existing pattern is archaic, as it's the way C code is written and not the way C++ is. We have been updating our idioms to C++ ones when the opportunity presents. This PR is exactly such an opportunity.